apis/src/components/organisms/analysis/tree.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::VecDeque;
+
+use crate::providers::game_state::{GameState, GameStateSignal};
+use hive_lib::{GameType, History, State};
+use leptos::*;
+use serde::{Deserialize, Serialize};
+use tree_ds::prelude::{Node, Tree};
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct TreeNode {
+    pub turn: usize,
+    pub piece: String,
+    pub position: String,
+}
+
+#[derive(Clone)]
+pub struct AnalysisSignal(pub RwSignal<Option<AnalysisTree>>);
+#[derive(Clone, Default, Serialize, Deserialize)]
+pub struct AnalysisTree {
+    pub current_node: Option<usize>, // The node that is currently being viewed
+    pub tree: Tree<usize, TreeNode>,
+}
+
+impl AnalysisTree {
+    pub fn from_state(game_state: GameStateSignal) -> Self {
+        let gs = game_state.signal.get_untracked();
+        let mut tree = Tree::new(Some("analysis"));
```

Comment:
Maybe we want to name the Tree with "white.username-black.username-date-outcome?` So that we can use that as file name/general title for the analysis board.